Database migration fails when upgrading to 2.6.1

### Bug Description

When upgrading to 2.6.1 with existing `datasetconfig` objects, the migration can fail when migrating existing datasets:
```
  File "/usr/local/lib/python3.9/site-packages/fides/api/ctl/migrations/versions/216cdc7944f1_add_datasetconfig_ctl_datasets_fk.py", line 69, in upgrade
    appended_meta: Dict = dataset["meta"] or {}
    │                     └ {'name': 'Adobe Campaign Dataset', 'fides_key': 'sdfgs', 'collections': [{'name': 'profile', 'fields': [{'name': 'PKey', 'fid...
    └ {'fides_source': 'auto-migrated from datasetconfig.dataset'}

KeyError: 'meta'
```

#### Steps to Reproduce

1. Run `fides` on v2.5.x
2. Configure a connector of some kind (SaaS, database, etc.) and attach a dataset
3. Upgrade to v2.6.1

### Expected behavior

Migration should succeed smoothly and transition the existing `datasetconfig` rows to `ctl_dataset` rows

### Example stack trace

```
2023-02-06 14:10:39.482 [ERROR] (database:configure_db:117): Unable to configure database: KeyError: 'meta'
2023-02-06 14:10:39.483 [ERROR] (database:configure_db:118): 'meta'
Traceback (most recent call last):

  File "<redacted"", line 8, in <module>
    sys.exit(main())
    │   │    └ <function main at 0x4032aa8dc0>
    │   └ <built-in function exit>
    └ <module 'sys' (built-in)>
  File "<redacted"", line 64, in main
    start_webserver()
    └ <function start_webserver at 0x403eb83ca0>
  File "/usr/local/lib/python3.9/site-packages/fides/api/main.py", line 377, in start_webserver
    server.run()
    │      └ <function Server.run at 0x40059343a0>
    └ <uvicorn.server.Server object at 0x408b682d00>
  File "/usr/local/lib/python3.9/site-packages/uvicorn/server.py", line 60, in run
    return asyncio.run(self.serve(sockets=sockets))
           │       │   │    │             └ None
           │       │   │    └ <function Server.serve at 0x4005934430>
           │       │   └ <uvicorn.server.Server object at 0x408b682d00>
           │       └ <function run at 0x40022c73a0>
           └ <module 'asyncio' from '/usr/local/lib/python3.9/asyncio/__init__.py'>
  File "/usr/local/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
           │    │                  └ <coroutine object Server.serve at 0x408b6af140>
           │    └ <method 'run_until_complete' of 'uvloop.loop.Loop' objects>
           └ <uvloop.Loop running=True closed=False debug=False>
  File "/usr/local/lib/python3.9/site-packages/uvicorn/lifespan/on.py", line 84, in main
    await app(scope, self.receive, self.send)
          │   │      │    │        │    └ <function LifespanOn.send at 0x408b911af0>
          │   │      │    │        └ <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>
          │   │      │    └ <function LifespanOn.receive at 0x408b911b80>
          │   │      └ <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>
          │   └ {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'app': <fastapi.applications.FastAPI object at 0x4002...
          └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x408b688850>
  File "/usr/local/lib/python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 75, in __call__
    return await self.app(scope, receive, send)
                 │    │   │      │        └ <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>>
                 │    │   │      └ <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>>
                 │    │   └ {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'app': <fastapi.applications.FastAPI object at 0x4002...
                 │    └ <fastapi.applications.FastAPI object at 0x40021052b0>
                 └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x408b688850>
  File "/usr/local/lib/python3.9/site-packages/fastapi/applications.py", line 269, in __call__
    await super().__call__(scope, receive, send)
                           │      │        └ <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>>
                           │      └ <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>>
                           └ {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'app': <fastapi.applications.FastAPI object at 0x4002...
  File "/usr/local/lib/python3.9/site-packages/starlette/applications.py", line 124, in __call__
    await self.middleware_stack(scope, receive, send)
          │    │                │      │        └ <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>>
          │    │                │      └ <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>>
          │    │                └ {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'app': <fastapi.applications.FastAPI object at 0x4002...
          │    └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x403ebb0be0>
          └ <fastapi.applications.FastAPI object at 0x40021052b0>
  File "/usr/local/lib/python3.9/site-packages/starlette/middleware/errors.py", line 149, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>>
          │    │   │      └ <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>>
          │    │   └ {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'app': <fastapi.applications.FastAPI object at 0x4002...
          │    └ <starlette.middleware.base.BaseHTTPMiddleware object at 0x403ebb0b80>
          └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x403ebb0be0>
  File "/usr/local/lib/python3.9/site-packages/starlette/middleware/base.py", line 24, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>>
          │    │   │      └ <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>>
          │    │   └ {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'app': <fastapi.applications.FastAPI object at 0x4002...
          │    └ <starlette.middleware.base.BaseHTTPMiddleware object at 0x403ebb0b20>
          └ <starlette.middleware.base.BaseHTTPMiddleware object at 0x403ebb0b80>
  File "/usr/local/lib/python3.9/site-packages/starlette/middleware/base.py", line 24, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>>
          │    │   │      └ <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>>
          │    │   └ {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'app': <fastapi.applications.FastAPI object at 0x4002...
          │    └ <starlette.middleware.cors.CORSMiddleware object at 0x403ebb0af0>
          └ <starlette.middleware.base.BaseHTTPMiddleware object at 0x403ebb0b20>
  File "/usr/local/lib/python3.9/site-packages/starlette/middleware/cors.py", line 76, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>>
          │    │   │      └ <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>>
          │    │   └ {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'app': <fastapi.applications.FastAPI object at 0x4002...
          │    └ <slowapi.middleware.SlowAPIMiddleware object at 0x403ebb0a60>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x403ebb0af0>
  File "/usr/local/lib/python3.9/site-packages/starlette/middleware/base.py", line 24, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>>
          │    │   │      └ <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>>
          │    │   └ {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'app': <fastapi.applications.FastAPI object at 0x4002...
          │    └ <starlette.exceptions.ExceptionMiddleware object at 0x403ebb0a00>
          └ <slowapi.middleware.SlowAPIMiddleware object at 0x403ebb0a60>
  File "/usr/local/lib/python3.9/site-packages/starlette/exceptions.py", line 69, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>>
          │    │   │      └ <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>>
          │    │   └ {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'app': <fastapi.applications.FastAPI object at 0x4002...
          │    └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x403ebb0790>
          └ <starlette.exceptions.ExceptionMiddleware object at 0x403ebb0a00>
  File "/usr/local/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>>
          │    │   │      └ <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>>
          │    │   └ {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'app': <fastapi.applications.FastAPI object at 0x4002...
          │    └ <fastapi.routing.APIRouter object at 0x40020f00d0>
          └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x403ebb0790>
  File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 659, in __call__
    await self.lifespan(scope, receive, send)
          │    │        │      │        └ <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>>
          │    │        │      └ <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0x408b688880>>
          │    │        └ {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'app': <fastapi.applications.FastAPI object at 0x4002...
          │    └ <function Router.lifespan at 0x4004f93c10>
          └ <fastapi.routing.APIRouter object at 0x40020f00d0>
  File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 635, in lifespan
    async with self.lifespan_context(app):
               │    │                └ <fastapi.applications.FastAPI object at 0x40021052b0>
               │    └ <starlette.routing._DefaultLifespan object at 0x4005396fd0>
               └ <fastapi.routing.APIRouter object at 0x40020f00d0>
  File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 530, in __aenter__
    await self._router.startup()
          │    │       └ <function Router.startup at 0x4004f93af0>
          │    └ <fastapi.routing.APIRouter object at 0x40020f00d0>
          └ <starlette.routing._DefaultLifespan object at 0x4005396fd0>
  File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 612, in startup
    await handler()
          └ <function setup_server at 0x403eb83d30>
  File "/usr/local/lib/python3.9/site-packages/fides/api/main.py", line 237, in setup_server
    await configure_db(CONFIG.database.sync_database_uri)
          │            │      │        └ '<redacted"'
          │            │      └ DatabaseSettings(<redacted>...
          │            └ FidesConfig(test_mode=False, is_test_mode=False, hot_reloading=False, dev_mode=False, oauth_instance=None, admin_ui=AdminUISe...
          └ <function configure_db at 0x4009955e50>
> File "/usr/local/lib/python3.9/site-packages/fides/api/ctl/database/database.py", line 108, in configure_db
    await init_db(database_url)
          │       └ '<redacted>'
          └ <function init_db at 0x4009955c10>
  File "/usr/local/lib/python3.9/site-packages/fides/api/ctl/database/database.py", line 53, in init_db
    upgrade_db(alembic_config)
    │          └ <alembic.config.Config object at 0x408b9180d0>
    └ <function upgrade_db at 0x4009955b80>
  File "/usr/local/lib/python3.9/site-packages/fides/api/ctl/database/database.py", line 44, in upgrade_db
    command.upgrade(alembic_config, revision)
    │       │       │               └ 'head'
    │       │       └ <alembic.config.Config object at 0x408b9180d0>
    │       └ <function upgrade at 0x40098ead30>
    └ <module 'alembic.command' from '/usr/local/lib/python3.9/site-packages/alembic/command.py'>
  File "/usr/local/lib/python3.9/site-packages/alembic/command.py", line 322, in upgrade
    script.run_env()
    │      └ <function ScriptDirectory.run_env at 0x40099238b0>
    └ <alembic.script.base.ScriptDirectory object at 0x408b918040>
  File "/usr/local/lib/python3.9/site-packages/alembic/script/base.py", line 569, in run_env
    util.load_python_file(self.dir, "env.py")
    │    │                │    └ '/usr/local/lib/python3.9/site-packages/fides/api/ctl/database/../migrations'
    │    │                └ <alembic.script.base.ScriptDirectory object at 0x408b918040>
    │    └ <function load_python_file at 0x4009829040>
    └ <module 'alembic.util' from '/usr/local/lib/python3.9/site-packages/alembic/util/__init__.py'>
  File "/usr/local/lib/python3.9/site-packages/alembic/util/pyfiles.py", line 94, in load_python_file
    module = load_module_py(module_id, path)
             │              │          └ '/usr/local/lib/python3.9/site-packages/fides/api/ctl/database/../migrations/env.py'
             │              └ 'env_py'
             └ <function load_module_py at 0x40098290d0>
  File "/usr/local/lib/python3.9/site-packages/alembic/util/pyfiles.py", line 110, in load_module_py
    spec.loader.exec_module(module)  # type: ignore
    │    │      │           └ <module 'env_py' from '/usr/local/lib/python3.9/site-packages/fides/api/ctl/database/../migrations/env.py'>
    │    │      └ <function _LoaderBasics.exec_module at 0x400202b550>
    │    └ <_frozen_importlib_external.SourceFileLoader object at 0x408b918310>
    └ ModuleSpec(name='env_py', loader=<_frozen_importlib_external.SourceFileLoader object at 0x408b918310>, origin='/usr/local/lib...
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/usr/local/lib/python3.9/site-packages/fides/api/ctl/database/../migrations/env.py", line 86, in <module>
    run_migrations_online()
    └ <function run_migrations_online at 0x408b911ee0>
  File "/usr/local/lib/python3.9/site-packages/fides/api/ctl/database/../migrations/env.py", line 80, in run_migrations_online
    context.run_migrations()
    │       └ <function run_migrations at 0x40098e7700>
    └ <module 'alembic.context' from '/usr/local/lib/python3.9/site-packages/alembic/context.py'>
  File "<string>", line 8, in run_migrations
  File "/usr/local/lib/python3.9/site-packages/alembic/runtime/environment.py", line 853, in run_migrations
    self.get_context().run_migrations(**kw)
    │    │                              └ {}
    │    └ <function EnvironmentContext.get_context at 0x40098e4d30>
    └ <alembic.runtime.environment.EnvironmentContext object at 0x408b918190>
  File "/usr/local/lib/python3.9/site-packages/alembic/runtime/migration.py", line 623, in run_migrations
    step.migration_fn(**kw)
    │    │              └ {}
    │    └ <function upgrade at 0x408bae3820>
    └ RevisionStep('216cdc7944f1', is_upgrade=True)
  File "/usr/local/lib/python3.9/site-packages/fides/api/ctl/migrations/versions/216cdc7944f1_add_datasetconfig_ctl_datasets_fk.py", line 69, in upgrade
    appended_meta: Dict = dataset["meta"] or {}
    │                     └ {'name': 'Adobe Campaign Dataset', 'fides_key': 'sdfgs', 'collections': [{'name': 'profile', 'fields': [{'name': 'PKey', 'fid...
    └ {'fides_source': 'auto-migrated from datasetconfig.dataset'}

KeyError: 'meta'
```

### Environment

* Version: 2.6.1

--- a/src/fides/api/ctl/migrations/versions/216cdc7944f1_add_datasetconfig_ctl_datasets_fk.py
+++ b/src/fides/api/ctl/migrations/versions/216cdc7944f1_add_datasetconfig_ctl_datasets_fk.py
@@ -66,7 +66,7 @@ def upgrade():
 
         new_ctl_dataset_id: str = "ctl_" + str(uuid.uuid4())
         # Stashing extra text into the "meta" column so we can use this to downgrade if needed
-        appended_meta: Dict = dataset.get("meta", {})
+        appended_meta: Dict = dataset.get("meta") or {}
         appended_meta["fides_source"] = AUTO_MIGRATED_STRING
 
         validated_dataset: Dict = Dataset(

